marconi-chain-index/src/Marconi/ChainIndex/Experimental/Runner.hs

```diff
@@ -44,47 +59,77 @@ If you want to start several indexers, use @runIndexers@.
 -}
 runIndexer
   :: ( Core.IsIndex (ExceptT Core.IndexerError IO) (WithDistance BlockEvent) indexer
+     , Core.IsSync (ExceptT Core.IndexerError IO) (WithDistance BlockEvent) indexer
      , Core.Closeable IO indexer
      )
   => Trace IO Text
+  -> SecurityParam
   -> RetryConfig
   -> FilePath
   -> C.NetworkId
   -> C.ChainPoint
   -> indexer (WithDistance BlockEvent)
   -> IO ()
-runIndexer trace retryConfig socketPath networkId _startingPoint indexer = do
+runIndexer trace securityParam retryConfig socketPath networkId preferedStartingPoint indexer = do
   withNodeConnectRetry trace retryConfig socketPath $ do
-    securityParam <- Utils.toException $ Utils.querySecurityParam @Void networkId socketPath
+    (lastSyncPointE :: Either Core.IndexerError (Core.Point BlockEvent)) <-
+      runExceptT $ Core.lastSyncPoint indexer
+    lastSyncPoint <-
+      case lastSyncPointE of
+        Left e -> throwIO e
+        Right r -> pure r
+    Trace.logInfo trace $
+      PP.renderStrict $
+        PP.layoutPretty PP.defaultLayoutOptions $
+          PP.pretty $
+            LastSyncPointLog lastSyncPoint
+    startingPoint <-
+      case preferedStartingPoint of
+        C.ChainPointAtGenesis -> pure lastSyncPoint
+        cliChainPoint -> pure cliChainPoint
     eventQueue <- STM.newTBQueueIO $ fromIntegral securityParam
     cBox <- Concurrent.newMVar indexer
     let runChainSyncStream =
           withChainSyncBlockEventStream
             socketPath
             networkId
-            [Core.genesis]
+            [startingPoint]
             (mkEventStream eventQueue . chainSyncEventStreamLogging trace)
         whenNoIntersectionFound NoIntersectionFound =
           Trace.logError trace $
-            Pretty.renderStrict $
-              Pretty.layoutPretty
-                Pretty.defaultLayoutOptions
-                "No intersection found"
-    void $ runChainSyncStream `catch` whenNoIntersectionFound
-    Core.processQueue eventQueue cBox
+            PP.renderStrict $
+              PP.layoutPretty PP.defaultLayoutOptions $
+                PP.pretty NoIntersectionFoundLog
+    Async.concurrently_
+      (void $ runChainSyncStream `catch` whenNoIntersectionFound)
+      (Core.processQueue eventQueue cBox)
 
--- | Run several indexers under a unique coordinator
+{- | Run several indexers under a unique coordinator
+TODO Remove? The user can just call `mkCoordinator` himself.
```

Comment:
@berewt What do you think?

Reply:
I initially introduced this function for consistency with the previous API, I like the idea of being able to start a bunch of workers without having to deal with the coordinator but it's probably to naive and impossible to combine easily with the resuming, so it's probably better to drop it.